src/coreComponents/mesh/mpiCommunications/CommunicationTools.cpp

```diff
@@ -947,5 +947,70 @@ void CommunicationTools::synchronizeFields( const std::map< string, string_array
   synchronizeUnpack( mesh, neighbors, icomm, onDevice );
 }
 
+void CommunicationTools::synchronizeFields2( std::vector< NeighborCommunicator::SyncFieldsID > const & fieldsTobeSync,
+                                             MeshLevel & mesh,
+                                             std::vector< NeighborCommunicator > & neighbors,
+                                             bool onDevice )
+{
+  MPI_iCommData icomm( getCommID() );
+  icomm.resize( neighbors.size() );
+  synchronizePackSendRecvSizes2( fieldsTobeSync, mesh, neighbors, icomm, onDevice );
+  synchronizePackSendRecv2( fieldsTobeSync, mesh, neighbors, icomm, onDevice );
+  synchronizeUnpack( mesh, neighbors, icomm, onDevice );
+}
+
+void CommunicationTools::synchronizePackSendRecv2( std::vector< NeighborCommunicator::SyncFieldsID > const & fieldsTobeSync,
+                                                   MeshLevel & mesh,
+                                                   std::vector< NeighborCommunicator > & neighbors,
+                                                   MPI_iCommData & icomm,
+                                                   bool onDevice )
+{
+  GEOSX_MARK_FUNCTION;
+  parallelDeviceEvents events;
+  asyncPack2( fieldsTobeSync, mesh, neighbors, icomm, onDevice, events );
+  asyncSendRecv( neighbors, icomm, onDevice, events );
+}
+
+void CommunicationTools::synchronizePackSendRecvSizes2( std::vector< NeighborCommunicator::SyncFieldsID > const & fieldsTobeSync,
+                                                        MeshLevel & mesh,
+                                                        std::vector< NeighborCommunicator > & neighbors,
+                                                        MPI_iCommData & icomm,
+                                                        bool onDevice )
+{
+  GEOSX_MARK_FUNCTION;
+  icomm.resize( neighbors.size() );
```

Comment:
So you removed `  icomm.fieldNames().insert( fieldNames.begin(), fieldNames.end() );`?

Reply:
For now...I was trying to understand if that field was used anywhere.

Reply:
for now, this is not done. I am adding a piece at a time.